schemas/2016-03-01/Microsoft.ServiceFabric.json

```diff
@@ -0,0 +1,611 @@
+            },
+            "DurabilityLevel": {
+              "enum": [
+                "Bronze",
+                "Silver",
+                "Gold",
+                "Platinum"
+              ]
+            },
+            "VMInstanceCount": {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/numberOrExpression",
+              "minimum": 1,
+              "maximum": 2147483647
+            },
+            "Name": {
+              "type": "string",
+              "minLength": 1
+            },
+            "PlacementProperties": {
+              "oneOf": [
+                {
+                  "type": "object",
+                  "additionalProperties": {
+                    "type": "string"
+                  }
+                },
+                {
+                  "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                }
+              ]
+            },
+            "Capacities": {
+              "oneOf": [
+                {
+                  "type": "object",
+                  "additionalProperties": {
+                    "type": "string"
+                  }
+                },
+                {
+                  "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                }
+              ]
+            },
+            "ClientConnectionEndpointPort": {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/numberOrExpression"
+            },
+            "HttpGatewayEndpointPort": {
+              "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/numberOrExpression"
+            },
+            "ApplicationPorts": {
+              "oneOf": [
+                {
+                  "type": "object",
+                  "properties": {
+                    "StartPort": {
+                      "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/numberOrExpression"
+                    },
+                    "EndPort": {
+                      "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/numberOrExpression"
+                    }
+                  },
+                  "required": [
+                    "StartPort",
+                    "EndPort"
+                  ]
+                },
+                {
+                  "$ref": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
+                }
+              ]
+            },
+            "EphemeralPorts": {
+              "$ref": "#/definitions/NodeTypes/oneOf/0/properties/ApplicationPorts"
```

Comment:
can we create a definition for "ApplicationPorts" under "definitions", and refer to it from these places?